packages/integration-react/src/MerkurComponent.jsx

```diff
@@ -191,26 +174,40 @@ export default class MerkurComponent extends React.Component {
 
     this._widget.unmount();
     this._widget = null;
+
+    if (this.props.onWidgetUnmounted) {
+      this.props.onWidgetUnmounted();
+    }
   }
 
-  _tryCreateWidget() {
+  async _tryCreateWidget() {
     if (!this.props.widgetProperties || this._widget) {
       return;
     }
 
-    this._loadAssets(() => {
-      const merkur = getMerkur();
+    await this._loadScriptAssets();
 
-      try {
-        merkur.create(this.props.widgetProperties).then((widget) => {
-          this._widget = widget;
-          widget.mount();
-        });
-      } catch (_) {
-        if (this.props.debug) {
-          console.warn(_);
-        }
+    const merkur = getMerkur();
+
+    try {
+      this._widget = await merkur.create(this.props.widgetProperties);
+      this._widget.mount();
+
+      if (typeof this.props.onWidgetMounted === 'function') {
+        this.props.onWidgetMounted();
```

Comment:
I think that the onWidgetMounted method should get widget as argument.

Reply:
Passing `this._widget` to the function call